src/silx/gui/hdf5/Hdf5TreeModel.py

```diff
@@ -557,7 +565,7 @@ def synchronizeIndex(self, index):
     def h5pyObjectRow(self, h5pyObject):
         for row in range(self.__root.childCount()):
             item = self.__root.child(row)
-            if item.obj == h5pyObject:
```

Comment:
I feel like it's not a good idea.
Checking the values and chechink the hash is just not the same thing.
If you have implemented `__hash__` it is already checked by the equal.

Reply:
This is being reworked in #3665